src/stan_math_backend/Locations.ml

```diff
@@ -46,8 +46,10 @@ let pp_globals ppf location_list =
   let location_list =
     List.map
       ~f:(fun x ->
-        if x = no_span then "Found before start of program."
-        else "At " ^ string_of_location_span x )
+        " ("
+        ^ ( if x = no_span then "found before start of program"
```

Comment:
Is this the behaviour that we would like? That means that we should probably never insert no_span ourselves? For instance, AST_to_mir and optimizations should never insert a no_span that ends up being the error location, because we wouldn't really expect those errors to correspond to being triggered before the start of the program.

Reply:
I don't think I changed any behavior here - this is what you had before just with parenthesis added around the outside to match the existing compiler's format, right? Or did I mess that up?

Reply:
Agreed. You didn't change the behaviour. I put that in, but I'm still not sure it's right.
Probably it's OK for now.

Reply:
I need to see some examples but off the cuff I'd say if there's a no_span we just don't emit any location updates at all.

Reply:
That's actually a great idea.

Reply:
Added it to this PR as a separate commit.